cmd/mysterium_client/command_run/command.go

```diff
@@ -68,14 +71,24 @@ func (cmd *CommandRun) Run(options CommandOptions) (err error) {
 		return err
 	}
 
+	apiEndpoints := tequilapi.NewApiEndpoints()
+	//TODO additional endpoint registration can go here i.e apiEndpoints.GET("/path", httprouter.Handle function)
+
+	cmd.httpApiServer, err = tequilapi.NewServer(options.TequilaApiAddress, options.TequilaApiPort, apiEndpoints)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
 func (cmd *CommandRun) Wait() error {
+	cmd.httpApiServer.Wait()
```

Comment:
This wait makes `cmd.vpnClient.Wait()` statement unreachable.
See in `cmd/mysterium_fake/mysterium_fake.go` how to handle 2 concurrent waits

Reply:
Will fix

Reply:
Rewritten with error channels and error returning on it.